packages/colorpickers/src/elements/ColorSwatch/index.tsx

```diff
@@ -0,0 +1,109 @@
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { forwardRef, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { ThemeContext } from 'styled-components';
+import { Tooltip } from '@zendeskgarden/react-tooltips';
+import { useGrid } from '@zendeskgarden/container-grid';
+import { useId } from '@zendeskgarden/container-utilities';
+import { StyledSwatchButton, StyledCheckIcon, StyledCell, StyledColorSwatch } from '../../styled';
+
+export interface ILabeledColor {
+  value: string;
+  label: string;
+}
+
+export interface IColorSwatchProps {
+  /** Enables wrapped keyboard navigation */
+  isWrapped?: boolean;
+  /** Sets the two-dimension array of labeled HEX and RGB/A string colors */
+  colors: ILabeledColor[][];
+  /** Sets the focused row index in a controlled color swatch */
+  rowIndex?: number;
+  /** Sets the focused column index in a controlled color swatch */
+  colIndex?: number;
+  /** Sets the selected row index in a controlled color swatch */
+  selectedRowIndex?: number;
+  /** Sets the selected column index in a controlled color swatch */
+  selectedColIndex?: number;
+  /** Sets the default focused row index in a uncontrolled color swatch */
+  defaultRowIndex?: number;
+  /** Sets the default focused column index in a uncontrolled color swatch */
+  defaultColIndex?: number;
+  /** Sets the default selected row index in a uncontrolled color swatch */
+  defaultSelectedRowIndex?: number;
+  /** Sets the default selected column index in a uncontrolled color swatch */
+  defaultSelectedColIndex?: number;
+  /** Handles color swatch changes */
+  onChange?: (rowIndex: number, colIndex: number) => void;
+  /** Handles color swatch select event */
+  onSelect?: (rowIndex: number, colIndex: number) => void;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLTableElement>
+ */
+export const ColorSwatch = forwardRef<HTMLTableElement, IColorSwatchProps>(
+  ({ colors, isWrapped, ...props }, ref) => {
+    const { rtl } = useContext(ThemeContext);
+    const { getGridCellProps } = useGrid({
+      rtl,
+      matrix: colors,
+      selection: true,
+      wrap: isWrapped,
+      idPrefix: useId(),
+      ...props
+    });
+
+    return (
+      <StyledColorSwatch ref={ref}>
+        <tbody>
+          {colors.map((row: ILabeledColor[], rowIdx: number) => (
+            <tr key={row[0].value}>
+              {row.map((color: ILabeledColor, colIdx: number) => {
+                const { label, value } = color;
+                const gridCellProps = getGridCellProps({
+                  colIdx,
+                  rowIdx,
+                  type: 'button'
+                });
+
+                return (
+                  <StyledCell key={value}>
+                    <Tooltip content={label}>
+                      <StyledSwatchButton backgroundColor={value} {...gridCellProps}>
+                        {gridCellProps['aria-selected'] ? <StyledCheckIcon color={value} /> : null}
```

Comment:
It looks like the `CheckIcon` is persistent in the DOM and its opacity is transitioned from `0` and `1`. Since the checkmark for color chips are unmounted & mounted depending on selection, `react-transition-group` can assist in the transition. How does this approach sound?

Reply:
We should resist using `react-transition-group` when simple CSS opacity will suffice. In order to do this, the `<StyledCheckIcon>` remains persistent, but is styled conditionally. Similar to the component treatment I linked to.